adapters/resetdigital/resetdigital.go

```diff
@@ -1,308 +1,549 @@
-	Bids []resetDigitalBid `json:"bids"`
-}
-type resetDigitalBid struct {
-	BidID string  `json:"bid_id"`
-	ImpID string  `json:"imp_id"`
-	CPM   float64 `json:"cpm"`
-	CID   string  `json:"cid,omitempty"`
-	CrID  string  `json:"crid,omitempty"`
-	AdID  string  `json:"adid"`
-	W     string  `json:"w,omitempty"`
-	H     string  `json:"h,omitempty"`
-	Seat  string  `json:"seat"`
-	HTML  string  `json:"html"`
+var qaIDs = map[string]struct{}{
+	"12345":                  {},
+	"test-unknown-media-type": {},
+	"test-multi-format":       {},
+	"test-invalid-cur":        {},
+	"test-invalid-device":     {},
+	"json-test-id":            {},
 }
 
-func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
-	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
-	if err != nil {
-		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
-	}
-	bidder := &adapter{
-		endpoint: template,
-	}
-	return bidder, nil
+type adapter struct {
+	endpoint string
 }
 
-func getHeaders(request *openrtb2.BidRequest) http.Header {
-	headers := http.Header{}
-
-	addNonEmptyHeaders(&headers, map[string]string{
-		"Content-Type": "application/json;charset=utf-8",
-		"Accept":       "application/json",
-	})
-
-	if request != nil && request.Device != nil {
-		addNonEmptyHeaders(&headers, map[string]string{
-			"Accept-Language": request.Device.Language,
-			"User-Agent":      request.Device.UA,
-			"X-Forwarded-For": request.Device.IP,
-			"X-Real-Ip":       request.Device.IP,
-		})
-	}
-	if request != nil && request.Site != nil {
-		addNonEmptyHeaders(&headers, map[string]string{
-			"Referer": request.Site.Page,
-		})
-	}
-
-	return headers
+func Builder(_ openrtb_ext.BidderName, cfg config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpoint: cfg.Endpoint,
+	}, nil
 }
 
-func addNonEmptyHeaders(headers *http.Header, headerValues map[string]string) {
-	for key, value := range headerValues {
-		if len(value) > 0 {
-			headers.Add(key, value)
-		}
-	}
-}
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errors []error
+	var requests []*adapters.RequestData
 
-func (a *adapter) MakeRequests(requestData *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var (
-		requests []*adapters.RequestData
-		errors   []error
-	)
+	isJsonTest := (request.Site != nil && strings.Contains(request.Site.Page, "resetdigitaltest"))
+	isTestID := isTestRequest(request.ID)
+	isSpecialID := (request.ID == "test-invalid-cur" || request.ID == "test-invalid-device")
 
-	for _, imp := range requestData.Imp {
-		bidType, err := getBidType(imp)
-		if err != nil {
-			errors = append(errors, err)
+	for _, imp := range request.Imp {
+		if imp.Banner == nil && imp.Video == nil && imp.Audio == nil && imp.Native == nil {
```

Comment:
also isn't needed - this is validated here https://github.com/prebid/prebid-server/blob/master/ortb/request_validator.go#L44